### PR TITLE
1682 update remaining outdated npm dependencies

### PR DESCRIPTION
**Pull request by alexjanousekGSA**
## Description

Updated all outdated dependencies except for one (govuk-fe) that I will be recommending to be removed from the entire project because its a huge lift and we have our own US version of the component library. 

## TODO (optional)

* [x] govuk_frontend_toolkit
* [x] govuk-frontend (leaving for now)
* [x] gulp
* [x] morphdom
* [ ] replace rollup with webpack

## Security Considerations

* Govuk is being left for now because a update breaks the UI in the app due to api changes. Recommendation would be to get rid of this package all together since we have USWDS. 

## Todo
Finish updating rollup

### CODE REVIEW

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -23,6 +23,12 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
```
Comment (ccostino) [inline .github/workflows/checks.yml:27]: It looks like `v4` is the latest major version available for this action, we'll want to use that: https://github.com/actions/setup-node/releases/tag/v4.0.3 - unless there was an issue with it that you found?

File: .github/workflows/checks.yml [hunk 2]
```diff
@@ -23,6 +23,12 @@ jobs:
...
+        with:
+          node-version: "16.15.1"
```
Comment (ccostino) [inline .github/workflows/checks.yml:29]: Shouldn't this be the newest version that we switched to, `22.3.0`?

File: gulpfile.js [hunk 3]
```diff
@@ -27,7 +27,6 @@ plugins.uglify = require('gulp-uglify');
 const paths = {
   src: 'app/assets/',
   dist: 'app/static/',
-  templates: 'app/templates/',
```
Comment (ccostino) [inline gulpfile.js:30]: Just double checking here:  are we sure we don't need this anymore?

File: gulpfile.js [hunk 4]
```diff
@@ -150,8 +149,6 @@ const images = () => {
     paths.govuk_frontend + 'assets/images/**/*',
     paths.src + 'images/**/*',
     paths.src + 'img/**/*',
-    paths.template + 'assets/images/**/*'
```
Comment (ccostino) [inline gulpfile.js:153]: Just double checking here, too:  are we sure we don't need this anymore?

**Review by ccostino — COMMENTED** [review]
Thanks, @alexjanousekGSA!  There were a couple of things I wanted to check on with the GitHub Action and Node version; the other two things are just minor checks to make sure the changes are intentional.  Thanks!

File: .github/workflows/checks.yml (hunk 1 above)
Comment (alexjanousekGSA) [reply .github/workflows/checks.yml]: Good catch, updated.

File: .github/workflows/checks.yml (hunk 2 above)
Comment (alexjanousekGSA) [reply .github/workflows/checks.yml]: Another good one I missed when trying different things. Corrected.

File: gulpfile.js (hunk 3 above)
Comment (alexjanousekGSA) [reply gulpfile.js:30]: So leaving these gives the error: 

```[13:35:13] 'images' errored after 2.45 s
[13:35:13] Error: ENOENT: no such file or directory, scandir 'gsa/notifications-admin/undefinedassets/images'
[13:35:13] 'default' errored after 2.45 s
make: *** [bootstrap] Error 1
```

I manually looked for this directory and it no longer exists and running the project locally I see no missing images anywhere. Additionally, vscode suggests that that path doesnt exist either...oh wait, looks what i found...

If you notice, the path `templates` doesn't even match up to `paths.template`, its missing a `s` so it doesn't even work anyways and I still get the same error:

```[13:38:36] 'images' errored after 2.54 s
[13:38:36] Error: ENOENT: no such file or directory, scandir '/Users/ajanousek/gsa/notifications-admin/app/templates/assets/images'
[13:38:36] 'default' errored after 2.54 s
make: *** [bootstrap] Error 1
```

So I am confident that we can remove this code, considering it doesn't even work :)

Comment (ccostino) [reply gulpfile.js:30]: Fair enough, thank you for double checking and looking! :-)

**Review by ccostino — COMMENTED** [review]
Change-wise, this looks good to go for me now, thanks!

I know you tested it locally, so let's give this one more test run in our sandbox environment to make sure it deploys as expected!  [These are the instructions of how to do that](https://github.com/GSA/notifications-api/blob/main/docs/all.md#sandbox-environment), and please ask in Slack if you need any additional help or run into any trouble!

_Note that we'll have to deploy both the API and the Admin app (using this branch) to test fully as the admin site will expect the API to be up and running to function properly._

**Review by ccostino — APPROVED** [review]
Thank you, @alexjanousekGSA, and @jskinne3 for the assist! 🎉

**Pull request merged**